install.py

```diff
@@ -98,26 +101,117 @@ def copy_completion_scripts(dest_dir):
     return None
 
 
+def install(dest_dir, print_welcome=False):
+    """Install rez into the given directory.
+
+    Args:
+        dest_dir (str): Full path to the install directory.
+    """
+    print("installing rez to %s..." % dest_dir)
+
+    # create the virtualenv
+    create_environment(dest_dir)
+
+    # install rez from source
+    install_rez_from_source(dest_dir)
+
+    # patch the rez binaries
+    patch_rez_binaries(dest_dir)
+
+    # copy completion scripts into venv
+    completion_path = copy_completion_scripts(dest_dir)
+
+    # mark venv as production rez install. Do not remove - rez uses this!
+    _, _, _, venv_bin_dir = path_locations(dest_dir)
+    dest_bin_dir = os.path.join(venv_bin_dir, "rez")
+    validation_file = os.path.join(dest_bin_dir, ".rez_production_install")
+    with open(validation_file, 'w') as f:
+        f.write(_rez_version)
+
+    # done
+    if print_welcome:
+        print()
+        print("SUCCESS! To activate Rez, add the following path to $PATH:")
+        print(dest_bin_dir)
+
+        if completion_path:
+            print('')
+            shell = os.getenv('SHELL')
+
+            if shell:
+                shell = os.path.basename(shell)
+                ext = "csh" if "csh" in shell else "sh"  # Basic selection logic
+
+                print("You may also want to source the completion script (for %s):" % shell)
+                print("source {0}/complete.{1}".format(completion_path, ext))
+            else:
+                print("You may also want to source the relevant completion script from:")
+                print(completion_path)
+
+        print('')
+
+
 def install_rez_from_source(dest_dir):
     _, py_executable = get_py_venv_executable(dest_dir)
 
     # install via pip
     run_command([py_executable, "-m", "pip", "install", "."])
 
 
+def install_as_rez_package(repo_path):
+    """Installs rez as a rez package.
+
+    Note that this can be used to install new variants of rez into an existing
+    rez package (you may require multiple platform installations for example).
+
+    Args:
+        repo_path (str): Full path to the package repository to install into.
+    """
+    from tempfile import mkdtemp
+
+    # do a temp production (venv-based) rez install
+    tmpdir = mkdtemp(prefix="rez-install-")
+    install(tmpdir)
+
+    try:
```

Comment:
Maybe this temp install setup can be turned into a context so it can be re-used in the future:

<details><summary>temp_install context</summary>

```python
import contextlib


@contextlib.contextmanager
def temp_install(cleanup=True, print_welcome=False):
    """Do a temp production (venv-based) rez install.
    
    Args:
        cleanup (bool): Cleanup temp install when exiting context.
    """
    from tempfile import mkdtemp

    tmpdir = mkdtemp(prefix="rez-install-")
    install(tmpdir, print_welcome=print_welcome)

    try:
        yield tmpdir
    finally:
        if cleanup:
            try:
                shutil.rmtree(tmpdir)
            except Exception:
                pass


def install_as_rez_package(repo_path):
    """Installs rez as a rez package.
    
    Note that this can be used to install new variants of rez into an existing
    rez package (you may require multiple platform installations for example).
    
    Args:
        repo_path (str): Full path to the package repository to install into.
    """
    with temp_install() as tmpdir:
        # This extracts a rez package from the installation. See
        # rez.utils.installer.install_as_rez_package for more details.
        #
        args = (
            os.path.join(tmpdir, "bin", "python"), "-E", "-c",
            r"from rez.utils.installer import install_as_rez_package;"
            r"install_as_rez_package('%s')" % repo_path
        )
        print(subprocess.check_output(args))
```
</details>